ground/gcs/src/plugins/rawhid/rawhid.cpp

```diff
@@ -351,28 +278,29 @@ bool RawHID::open(OpenMode mode)
     return QIODevice::open(mode);
 }
 
+void RawHID::sendReadyRead()
+{
+    emit readyRead();
+}
+
 void RawHID::close()
 {
     if (!isOpen())
         return;
 
     RAW_HID_QXTLOG_DEBUG("RawHID: close()");
 
-    m_writeThread->stop();
     m_readThread->stop();
-
-    m_writeThread->wait(500);
-    m_readThread->wait(500);
-
-    m_writeThread->deleteLater();
+    m_readThread->wait();       /* Block indefinitely for read thread exit */
     m_readThread->deleteLater();
-
-    m_writeThread = NULL;
     m_readThread = NULL;
 
-    hid_close(m_handle);
+    m_writeThread->stop();
     m_handle = NULL;
 
+    m_writeThread = NULL;       /* Write thread is in charge of tearing itself and
+                                 * m_handle down */
+
     QIODevice::close();
```

Comment:
Yeah, so having `QIODevice::close` at the end can result in others trying to write after we've torn down the underlying USB stuff, especially since it emits `aboutToClose`.